src/compiler/checker.ts

```diff
@@ -8779,6 +8788,9 @@ namespace ts {
                     signature.unionSignatures ? getUnionType(map(signature.unionSignatures, getReturnTypeOfSignature), UnionReduction.Subtype) :
                     getReturnTypeFromAnnotation(signature.declaration!) ||
                     (nodeIsMissing((<FunctionLikeDeclaration>signature.declaration).body) ? anyType : getReturnTypeFromBody(<FunctionLikeDeclaration>signature.declaration));
+                if (signature.isOptionalCall) {

```

Comment:
I think there's enough of these fields now that `isOptionalCall`, `hasLiteralTypes`, and `hasRestParameter` should probably be combined into a `Flags` field of some kind.

Reply:
Related: `instantiateSignature` doesn't propagate this forward, so rather than a direct access, this needs to follow `.target`'s to check if the uninstantiated signature has a `isOptionalCall` field, no?

Reply:
That is correct.